packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -203,125 +203,133 @@ const makeRenderer = (
 	scripts: Array<string> = [],
 	enableServiceWorker: boolean,
 	cssLinks: ?Array<string>
-) => (request: HapiRequest, reply: HapiReply): Promise<RenderResult> => {
-	middleware = middleware || [];
+) => {
+	// set up a Promise that emits the resolved routes - this single Promise will
+	// be reused for all subsequent requests, so we're not resolving the routes repeatedly
+	// hooray performance
+	const routesPromise = resolveAllRoutes(routes);
```

Comment:
big indentation change because this promise gets set up once on server load (see code comment)